### DISCUSSION TITLE
Can kitty @ ls expose win.at_prompt property?

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#6670] [Ideas]
I would like to iterate over all kitty windows and execute a command only if the shell is idle and not running a command.  i.e., the shell is not running, say `vi` or any other command in the foreground.

For what I can tell, each window has an `at_prompt` property within Python, and I suppose this property may work _only_ with shell integration enabled.

Can we expose this within the kitty @ ls output?

An alternate would be to parse the output and check whether the foreground process is the shell itself.  But doing so is not all that easy using `jq` on the command line.

**Comment by kovidgoyal** [comment]
https://github.com/kovidgoyal/kitty/commit/fcce5c9a64f05042077713d2dc6b7bd06e57fc71
